x-pack/plugins/ml/public/application/components/scatterplot_matrix/scatterplot_matrix.tsx

```diff
@@ -0,0 +1,313 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useMemo, useEffect, useState, FC } from 'react';
+
+// There is still an issue with Vega Lite's typings with the strict mode Kibana is using.
+// @ts-ignore
+import { compile } from 'vega-lite/build-es5/vega-lite';
+import { parse, View, Warn } from 'vega';
+import { Handler } from 'vega-tooltip';
+
+import {
+  htmlIdGenerator,
+  EuiComboBox,
+  EuiComboBoxOptionOption,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiLoadingSpinner,
+  EuiSelect,
+  EuiSpacer,
+  EuiSwitch,
+  EuiText,
+} from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+
+import type { SearchResponse7 } from '../../../../common/types/es_client';
+
+import { ml } from '../../services/ml_api_service';
```

Comment:
as you created a new component, please use ml API service from the Kibana context

Reply:
Added to list of things to do in follow-up https://github.com/elastic/kibana/issues/84420